### ISSUE TITLE
Problem with nonlocal variables

### ISSUE THREAD

**Issue by da-woods** [numba/numba#938]
Simple example:

    from numba import jit
    def run():
      @jit
      def f(a):
        return n*a
      for n in range(5):
        print f(n)

prints five 0s. The pure python version correctly prints 0,1,4,9,16. This is using numba version 0.16.0, and python version 2.7.8.

I suspect it has to do with n being an integer (thus immutable) and so being re-assigned rather than modified in the loop. I don't know though.

I don't think this is a major problem - in most cases this can be fixed by passing another argument, and I could imagine the fix could be a lot slower. It's mostly just a notice that it doesn't quite work, and possibly it should be noted in the documentation somewhere.

**Comment by pitrou** [comment]
Yes, closure variables are frozen when the function is compiled. In this case n is 0 when the function is first compiled, hence what you are witnessing.
This is a limitation we're going to document once the documentation overhaul is finished.

**Comment by pitrou** [comment]
Closing as won't fix (we may want to do better one day, just not now :-)).
